obs-studio-server/source/nodeobs_service.h

```diff
@@ -176,7 +176,7 @@ class OBS_service
 	static void stopReplayBuffer(bool forceStop);
 	static void stopRecording(void);
 
-	static void releaseStreamingOutput(void);
+	static void releaseStreamingOutput(obs_output_t* output);
```

Comment:
Very odd that `releaseStreamingOutput` takes an `output` as an argument, I suggest we review the logic here since it is starting to get very wrong.

Reply:
Before my changes we had a data race, because the `streamingOutput` could be changed from 2 threads. This is  wrong.

My changes basically eliminates a data race, because the `releaseStreamingOutput `doesn't need to access `streamingOutput` global in order to delete it. And when `releaseStreamingOutput` is invoked, no one should store the output, cause it will be eventually deleted.

Reply:
Sounds good for the explanation. Please re work on the naming.